packages/es-components/src/components/util/useDisableBodyScroll.js

```diff
@@ -0,0 +1,23 @@
+import { useEffect } from 'react';
+
+export const useDisableBodyScroll = shouldDisableScroll => {
+  useEffect(() => {
+    const newStyle = 'overflow: hidden;';
+    const getCurrentStyle = () => document.body.getAttribute('style') || '';
+    const disableScroll = () =>
+      document.body.setAttribute('style', `${getCurrentStyle()} ${newStyle}`);
+    const enableScroll = () =>
+      document.body.setAttribute(
+        'style',
+        getCurrentStyle().replace(newStyle, '')
+      );
+
+    if (shouldDisableScroll) {
+      disableScroll();
+    } else {
+      enableScroll();
+    }
+
+    return enableScroll;
+  }, [shouldDisableScroll]);
```

Comment:
I was thinking about a way you might solve the issue you mentioned where it removes the body style when it renders, and I think I have a fairly clean solution:
```js
  useEffect(() => {
    const newStyle = 'overflow: hidden;';
    const getCurrentStyle = () => document.body.getAttribute('style') || '';
    let opened = false;
    const disableScroll = () => {
      opened = true;
      document.body.setAttribute('style', `${getCurrentStyle()} ${newStyle}`);
    }
    const enableScroll = () =>
      if (!opened) return;
      opened = false;
      document.body.setAttribute(
        'style',
        // this will only replace one instance of newStyle, which is what we want.
        getCurrentStyle().replace(newStyle, '')
      );

    if (shouldDisableScroll) {
      disableScroll();
    } else {
      enableScroll();
    }

    return enableScroll;
  }, [shouldDisableScroll]);

Reply:
I verified that your suggested approach works in all cases (when open state is controlled by  boolean prop and when Modal/SlidingPane is unmounted upon closing).